neurodamus/node.py

```diff
@@ -1865,50 +1866,54 @@ def _coreneuron_restore(self):
         self._sim_corenrn_write_config(corenrn_restore=True)
         self._sim_ready = True
 
-    # -
-    def _instantiate_simulation(self):
-        # Keep the initial RSS for the SHM file transfer calculations
-        self._initial_rss = SHMUtil.get_node_rss()
-        print_mem_usage()
-
-        self.load_targets()
-
-        # Check connection block configuration and raise warnings for overriding
-        # parameters
-        SimConfig.check_connections_configure(self._target_manager)
+    def n_cycles(self):
+        """Determine the number of model-building cycles
 
-        # Check if user wants to build the model in several steps (only for CoreNeuron)
+        It is based on configuration and system constraints.
+        """
         n_cycles = SimConfig.modelbuilding_steps
-
-        # Without multi-cycle, it's a trivial model build. sub_targets is False
+        # No multi-cycle. Trivial result, this is always possible
         if n_cycles == 1:
-            self._build_model()
-            return
+            return n_cycles
 
         target = self._target_manager.get_target(self._target_spec)
         target_name = self._target_spec.name
-        cell_count = target.gid_count()
-        logging.info("Simulation target: %s, Cell count: %d", target_name, cell_count)
+        max_cell_count = target.max_gid_count_per_population()
+        logging.info(
+            "Simulation target: %s, Max cell count per population: %d", target_name, max_cell_count
+        )
 
-        if SimConfig.use_coreneuron and cell_count / n_cycles < MPI.size and cell_count > 0:
+        if SimConfig.use_coreneuron and max_cell_count / n_cycles < MPI.size and max_cell_count > 0:
             # coreneuron with no. ranks >> no. cells
-            # need to assign fake gids to artificial cells in empty threads during module building
-            # fake gids start from max_gid + 1
+            # need to assign fake gids to artificial cells in empty threads
+            # during module building fake gids start from max_gid + 1
             # currently not support engine plugin where target is loaded later
-            max_num_cycles = int(cell_count / MPI.size) or 1
+            max_num_cycles = int(max_cell_count / MPI.size) or 1
             if n_cycles > max_num_cycles:
                 logging.warning(
                     "Your simulation is using multi-cycle without enough cells.\n"
                     "  => Number of cycles has been automatically set to the max: %d",
                     max_num_cycles,
                 )
                 n_cycles = max_num_cycles
+        return n_cycles
+
+    def _multicycle_build_model(self):
+        """Build the model iteratively over multiple cycles if required
 
+        Note: only relevant for coreNeuron
+        """
+        
+        n_cycles = self.n_cycles()
```

Comment:
I notice the attribute `self._n_cycles` which was forgotten in the pervious developments (don't know when it started).  It should be `self._n_cycles = self.n_cycles()`. 
The original commit is here
https://github.com/openbraininstitute/neurodamus/commit/6745c25ace8edf63045991d1d09e61e7fb12688d#diff-b3e8ca5ec7029b6ed9b0261d7de7443c95ac4e0a0e2e879ece996c35084798e7

Reply:
done